File: TMS150/lab1.m
clc;clf;clear
N=100;
X = normrnd(0,1,[1,N]);
hist(X)

X=sort(X);

phat=mle(X);

x=zeros(1,N);
y=zeros(1,N);
for i=1:N
    x(1,i)=(i-0.5)/N;
    y(i,1)=normcdf(X(1,i),phat(1),phat(2));
end

plot(x,y)

hold on;

plot(x,x)

% vi får då att P( D <= 0.136) = 0.95.

D=max(abs(normcdf(X,phat(1),phat(2))-(1:N)./N))
a=0.136

%%
clc;clf;clear
N=100;
X=gammarnd(2,2,[1,N])

File: TMS150/lab1.r
v <- rgamma(100,scale=2,shape=2)
hist(v,col = "blue",main = "hist gamma")

var = var(v) #MLE
mean = mean(v)

v = sort(v)

x = (1:100)/100
y = pnorm(v,mean,sqrt(var))


plot(x,y)
lines(c(0,1),c(0,1))


testSample = rnorm(100000,mean,sqrt(var))
D=max(abs(pnorm(v,mean,sqrt(var))-(1:100)/100))
print(D)
test = ks.test(v,testSample,alternative="two.sided")
test



